### PR TITLE
Fix the calculation of the histogram buckets and writing to the tensor in summary_db_writer

### PR DESCRIPTION
**Pull request by skambha**
[Issue: 1141](https://github.com/tensorflow/tensorboard/issues/1141): Incorrect data being written out for the histogram in the db

- The calculation of the histogram buckets is incorrect per the histogramProto defined in 	[summary.proto](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/framework/summary.proto)
- The data is getting clobbered in MigrateHistogram when writing it to a Tensor. This results in incorrect data being stored in the db

The pr provides a fix for this.

Testing: 
- Added a unit test

### CODE REVIEW

File: tensorflow/contrib/tensorboard/db/summary_db_writer.cc [hunk 1]
```diff
@@ -1182,14 +1182,19 @@ class SummaryDbWriter : public SummaryWriterInterface {
...
-    for (int i = 0; i < k; ++i) {
-      double left_edge = ((i - 1 >= 0) ? histo.bucket_limit(i - 1)
-                                       : std::numeric_limits<double>::min());
-      double right_edge = ((i + 1 < k) ? histo.bucket_limit(i + 1)
-                                       : std::numeric_limits<double>::max());
-      data(i + 0) = left_edge;
-      data(i + 1) = right_edge;
-      data(i + 2) = histo.bucket(i);
+    for (int i = 0, j = 0; i < k; ++i) {
+      // From summary.proto
```
Comment (nfelt) [inline tensorflow/contrib/tensorboard/db/summary_db_writer.cc]: Could you add here `// TODO(nickfelt): reconcile with TensorBoard's data_compat.py`?

File: tensorflow/contrib/tensorboard/db/summary_db_writer_test.cc [hunk 2]
```diff
@@ -100,6 +100,55 @@ class SummaryDbWriterTest : public ::testing::Test {
...
+    proto->add_bucket_limit(bucket_limits[i]);
+    proto->add_bucket(bucket[i]);
+  }
+  TF_ASSERT_OK(writer_->WriteEvent(std::move(e)));
+  TF_ASSERT_OK(writer_->Flush());
+  writer_->Unref();
+  writer_ = nullptr;
+
+  // Verify the data
+  string result = QueryString("SELECT data FROM Tensors");
```
Comment (nfelt) [inline tensorflow/contrib/tensorboard/db/summary_db_writer_test.cc]: Could you add here `// TODO(nickfelt): implement QueryTensor() to encapsulate this`?

**Review by nfelt — DISMISSED** [review]
Thank you very much for catching this and sending the fix!

Note that even with this change, I think there may still be some rendering issues in TensorBoard since TensorBoard itself doesn't expect to get DBL_MIN as a bucket boundary (in data_compat.py it instead uses the histogram's minimum seen value for the bottom bucket boundary).  However, I can look into resolving this discrepancy; please just add a TODO for me.

**Comment by skambha** [comment]
Thanks for the review.   Yes, thats right. I noticed that too in my experiments and had to hack/strip the dbl min in order to get the rendering to work.  I was thinking of opening an issue but if the todo's in code suffice, I can leave it at that. 

I have added the todo's per your request and that made your approval stale.  Can you please approve again.  Thank you.

**Review by nfelt — APPROVED** [review]
Thanks for adding the todos.  Perhaps what makes the most sense is to reopen the TensorBoard issue you filed and I'll update it when I can further fix that aspect.

**Comment by skambha** [comment]
That sounds good.  I have updated the title for that TensorBoard issue and reopened it, so we can use that.

**Comment by skambha** [comment]
Thank you @nfelt

**Pull request merged**